### PR TITLE
Add Motion2DComponent

### PR DESCRIPTION
**Pull request by tigleym**
For #28 

I think this work should be reviewed and merged after the `feature-hitbox2d` branch is merged into `cdsupina-dev` since there is already a lot of changes happening to the enemy component, entities, and related systems that should follow the same patterns introduced in `feature-hitbox2d`.

### CODE REVIEW

**Comment by tigleym** [comment]
@cdsupina The most relevant part of this patch that needs some discussion is splitting the collision handler system into two separate ones: `SpaceshipCollisionSystem` and `EnemyCollisionSystem`.

The biggest reason for my decision to do this is because we need to mutably borrow the `Motion2DComponent` storage for both the spaceship and enemy entities. Currently, this is done inside a nested for-loop, which will cause the borrow checker to fail because we're mutably borrowing for both entity types. 

I'm open to other approaches to solve this issue.

**Review by cdsupina — COMMENTED** [review]
I noted a few changes to the properties in Motion2DComponent. I think `max_speed` and `knockback_max_speed` should be included. I also had a few nits in the enemies.ron changes.

I think we should consider even adding an enum value to Motion2DComponent called something like `physics_type` where we can specify whether the motion is "Kinematic" or a "Rigidbody" (doesn't respond to collisions or does). For example, the normal enemies and player would be rigidbodies, but the Repeater boss parts are kinematic (they do not react to collisions).

I think that `physics_type` is outside the scope of this issue, but I think that it should be included in a future issue focusing on refactoring the collision physics and motion logic.

**Comment by cdsupina** [comment]
> 
> 
> @cdsupina The most relevant part of this patch that needs some discussion is splitting the collision handler system into two separate ones: `SpaceshipCollisionSystem` and `EnemyCollisionSystem`. This is the commit where this is implemented: [9dce8b6](https://github.com/amethyst/space_shooter_rs/commit/9dce8b60ad13256f65aea17bb6e13b9d7fa56203)
> 
> The biggest reason for my decision to do this is because we need to mutably borrow the `Motion2DComponent` storage for both the spaceship and enemy entities. Currently, this is done inside a nested for-loop, which will cause the borrow checker to fail because we're mutably borrowing for both entity types.
> 
> Let me know what you think about it. I'm open to other approaches to solve this issue!

This is exactly what I would have done to solve the issue. I agree with your decision to create two systems.

File: src/components/enemy.rs [hunk 1]
```diff
@@ -225,18 +158,6 @@ impl Fires for Enemy {
...
-        let mut rng = rand::thread_rng();
-        let rand_num = rng.gen_range(0, 2);
-
-        if rand_num == 1 {
-            self.current_velocity_x *= -1.0;
-            self.acceleration_x *= -1.0;
-        }
-    }
-}
-
```
[the remark this answers is not in the document]
Comment (tigleym) [reply src/components/enemy.rs:239]: I think `Spawnable` type methods are only used in `spawn_sprite_entity` and `spawn_animated_entity` now. I couldn't see any instances of either being used to create an enemy entity on the `cdsupina-dev` branch, so I figured we can remove it.

**Comment by tigleym** [comment]
Thanks for the review @cdsupina! I left a comment about removing the `Spawnable` trait implementation on the Enemy component. Let me know what you think!

File: src/components/enemy.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cdsupina) [reply src/components/enemy.rs:235]: Sounds good to me. I just didn't remember what `Spawnable' was used for.

**Comment by tigleym** [comment]
> I think we should consider even adding an enum value to Motion2DComponent called something like `physics_type` where we can specify whether the motion is "Kinematic" or a "Rigidbody" (doesn't respond to collisions or does). For example, the normal enemies and player would be rigidbodies, but the Repeater boss parts are kinematic (they do not react to collisions).
> 
> I think that `physics_type` is outside the scope of this issue, but I think that it should be included in a future issue focusing on refactoring the collision physics and motion logic.

Sounds good to me!

**Review by cdsupina — APPROVED** [review]
Looks good to me. I'm just going to make sure it runs on Windows and then merge.

**Pull request merged**